### PR TITLE
Fetch all points in time from indigo

### LINKED ISSUE #324: Lawlibrary must handle stub documents from Indigo

### LINKED ISSUE #342: As a legislation reader I want to see historical versions of a document
* indigo ingestor must import all points in time, in all languages

### PR DESCRIPTION
**Pull request by longhotsummer**
* fetch all points in time from indigo - closes #342 
* handle stubs from indigo - closes #324 
* stubs must use publication document as source and not store any content_html
* when clearing ingestor dates, queue the background task to update them
* source file downloading: show file type, decent filename
* store attachment sizes in db rather than fetching from storage
* logging

image

### CODE REVIEW

File: peachjam/views/documents.py [hunk 1]
```diff
@@ -57,8 +48,15 @@ class DocumentSourceView(DetailView):
 
     def render_to_response(self, context, **response_kwargs):
         if hasattr(self.object, "source_file") and self.object.source_file.file:
-            file = self.object.source_file.file.open()
-            return view_attachment(file)
+            source_file = self.object.source_file
+            file = source_file.file.open()
+            bytes = file.read()
+            response = HttpResponse(bytes, content_type=source_file.mimetype)
```
Comment (nickmwangemi) [inline peachjam/views/documents.py:54]: Assigning to a builtin variable. 
```suggestion
            file_bytes = file.read()
            response = HttpResponse(file_bytes, content_type=source_file.mimetype)
```

File: peachjam/views/documents.py [hunk 2]
```diff
@@ -57,8 +48,15 @@ class DocumentSourceView(DetailView):
...
+            response[
+                "Content-Disposition"
+            ] = f"inline; filename={source_file.filename_for_download()}"
+            response["Content-Length"] = str(len(bytes))
```
Comment (nickmwangemi) [inline peachjam/views/documents.py:58]:
```suggestion
            response["Content-Length"] = str(len(file_bytes))
```

**Review by nickmwangemi — CHANGES_REQUESTED** [review]
Tested the changes locally and they work well. Just that one minor change and merge conflict resolution and it should be good to go.

**Review by actlikewill — APPROVED** [review]

**Comment by longhotsummer** [comment]
> Tested the changes locally and they work well. Just that one minor change and merge conflict resolution and it should be good to go.

Nice catch, thanks. Committed your suggestions.

**Review by nickmwangemi — APPROVED** [review]

**Pull request merged**